miditok/tokenizations/remi_plus.py

```diff
@@ -0,0 +1,814 @@
+                    ):
+                        # Will loop over incoming tempo changes
+                        for tempo_change in self._current_midi_metadata[
+                            "tempo_changes"
+                        ][current_tempo_idx + 1 :]:
+                            # If this tempo change happened before the current moment
+                            if tempo_change.time <= note.start:
+                                current_tempo = tempo_change.tempo
+                                current_tempo_idx += (
+                                    1  # update tempo value (might not change) and index
+                                )
+                            else:  # <==> elif tempo_change.time > previous_tick:
+                                break  # this tempo change is beyond the current time step, we break the loop
+                    if nb_new_bars > 0:  # after the new Bar token
+                        # Position before the Tempo token
+                        pos_index = int((note.start % ticks_per_bar) / ticks_per_sample)
+                        events.append(
+                            Event(
+                                type="Position",
+                                value=pos_index,
+                                time=note.start,
+                                desc="TempoPosition",
+                            )
+                        )
+                        events.append(
+                            Event(
+                                type="Tempo",
+                                value=current_tempo,
+                                time=note.start,
+                                desc=note.start,
+                            )
+                        )
+
+                # (Chord)
+                if chord_results is not None:
+                    """
+                    chord_results: list of chord with ticks range
+                    [(start_tick, end_tick, chord_string)...]
+                    """
+                    pos_index = int((note.start % ticks_per_bar) / ticks_per_sample)
+                    if nb_new_bars > 0 and current_chord != "":
+                        # put chord token in every bar start
+                        events.append(
+                            Event(
+                                type="Position",
+                                value=pos_index,
+                                time=note.start,
+                                desc="ChordPosition",
+                            )
+                        )
+                        events.append(
+                            Event(
+                                type="Chord",
+                                value=current_chord,
+                                time=note.start,
+                                desc=note.start,
+                            )
+                        )
+                    elif (
+                        note.start > chord_results[current_chord_idx][0]
```

Comment:
Could be simplified with `chord_results[current_chord_idx][0] < note.start < chord_results[current_chord_idx][1]`